src/points/cardDefinitions.tsx

```diff
@@ -40,3 +49,44 @@ export default function useCardDefinitions(
     },
   }
 }
+
+export function useGetHistoryDefinition(): (history: ClaimHistory) => HistoryCardMetadata {
+  const { t } = useTranslation()
+  const tokensById = useSelector((state) =>
+    tokensByIdSelector(state, getSupportedNetworkIdsForSwap())
+  )
+
+  return (history: ClaimHistory) => {
+    // This pattern gives us static checking that we've exhausted all string union values
```

Comment:
I don't think it does check for exhaustiveness,
We need a `default` with `never` for this.
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking

Reply:
I believe this approach does check for exhaustiveness, albeit somewhat indirectly; removing a case statement causes the compiler to recognize the return value as potentially undefined/unassigned; this warning only goes away all possible cases are handled.

I'm not sure which pattern is more idiomatic; I was thinking we should go with whatever pattern gives us a more useful error message, though to be honest, neither of them give a _super_ actionable error at compile time... One added benefit of the pattern you mention is that it avoids needing `break` statements within the cases, which I like to avoid when possible, since it's easy to forget a `break` and have accidental fall-through. Will go with this pattern!

Reply:
What I tested is that it doesn't complain if we add a new value to the `claimActivities` array.
Which is important that it does complain if we do that, no?

Reply:
Oh interesting... I tested by removing a case from the switch statement and it did complain at me... which I would imagine is sort of equivalent to adding a new value to the `claimActivities` array... Definitely important that it complains in both of those examples. In any case, I've updated this to use the new pattern.